### ISSUE TITLE
Config exclusionary rule for extension_loaded

### ISSUE THREAD

**Issue by UndefinedOffset** [silverstripe/silverstripe-framework#9096]
## Affected Version

SilverStripe 4.4.1

## Description

It would nice if there was a way to exclude configuration sections based on whether or not an extension is loaded for example in our webbuilders-group/silverstripe-new-relic module we try to enable the Monolog handler for New Relic. However because of the order of the yaml config vs the php config the constant is not defined ([see here](https://github.com/webbuilders-group/silverstripe-new-relic/blob/2.1.1/_config/config.yml#L14-L19). The Monolog handler for New Relic throws an exception if the extension is not loaded. So if we could say do something like the below that would save this [kinda ugly work around](https://github.com/webbuilders-group/silverstripe-new-relic/commit/b21885a6f22f47f9edd62316dda6250d56f39f58#diff-0bcefa158fae7b420415084fad0ee875). On our development environments New Relic is not installed (and I would assume that's likely the case for most devs) but on live and staging it is.

```yml
---
Name: new_relic_error
After: "#new_relic"
Only:
    extensionloaded: "newrelic"
---
SilverStripe\Core\Injector\Injector:
    Psr\Log\LoggerInterface:
        calls:
            NewRelicHandlerNotice: [ pushHandler, [ '%$NewRelicHandlerNotice' ] ]
            NewRelicHandlerWarn: [ pushHandler, [ '%$NewRelicHandlerWarn' ] ]
            NewRelicHandlerError: [ pushHandler, [ '%$NewRelicHandlerError' ] ]
    NewRelicHandlerNotice:
        class: Monolog\Handler\NewRelicHandler
        constructor:
            - "notice"
    NewRelicHandlerWarn:
        class: Monolog\Handler\NewRelicHandler
        constructor:
            - "warning"
    NewRelicHandlerError:
        class: Monolog\Handler\NewRelicHandler
        constructor:
            - "error"
```

# Pull request
* [x] https://github.com/silverstripe/silverstripe-framework/pull/9106

**Comment by maxime-rainville** [comment]
When you say extension, do you mean like a `SilverStripe\Core\Extension`? If that's the case, it might not be possible because you'll probably hit a race condition ... you can't know if the extension you want is loaded until you're done reading all the configs.

If you mean a module, there's already a `moduleexists` exist rule you could use to exclude a YML section.

If someone wants to have a go at creating a PR with a new rule, I think the relevant logic is handled in [`SilverStripe\Core\Config\CoreConfigFactory::buildYamlTransformerForPath()`](https://github.com/silverstripe/silverstripe-framework/blob/909723f8e709a5d1720d84739a80e6a9cfa14b17/src/Core/Config/CoreConfigFactory.php#L132).

**Comment by kinglozzer** [comment]
@maxime-rainville I think in this case @UndefinedOffset is referring to PHP extensions: https://docs.newrelic.com/docs/agents/php-agent/installation/php-agent-installation-overview

@UndefinedOffset So at the simplest level, this would work by calling `extension_loaded('newrelic')`? Sounds simple enough, want to take a crack at implementing it? 😄

**Comment by UndefinedOffset** [comment]
ya as confusing as it is :) I am referring to PHP extensions, as for a pull I'll get on that later today :)

**Comment by UndefinedOffset** [comment]
Should it be against 4 or 4.4?

**Comment by kinglozzer** [comment]
@UndefinedOffset As it’s a new feature it should be against 4 for inclusion in 4.5

**Comment by maxime-rainville** [comment]
Fixed by #1459
